### PR TITLE
Fixes #16 Add possiblity to handle multipath.conf

### LINKED ISSUE #16: dsu with multipath
Hi,
I am thinking about implementing multipath.conf blacklist handling in this role, would you say this is a good place for this or should it be a separate role for changing the multipath config?

Background:
https://bugzilla.redhat.com/show_bug.cgi?id=1380272
if you use multipath you need to blacklist the devices dsu uses during upgrade otherwise they'll be in use by multipath

Greetings
Klaas

### PR DESCRIPTION
**Pull request by Klaas-**

### CODE REVIEW

File: tasks/multipathd.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    name: multipathd
+  register: service_multipathd
+
+- name: Add blacklist to multipath.conf
+  blockinfile:
+    block: |
+      blacklist {
+              devnode "^(ram|raw|loop|fd|md|dm-|sr|scd|st)[0-9]*"
+              devnode "^hd[a-z]*"
+              devnode "^cciss!c[0-9]d[0-9]*[p[0-9]*]"
```
Comment (martbhell) [inline tasks/multipathd.yml:19]: Is this cciss blacklist really needed?
Sounds HP specific to me

**Review by martbhell — COMMENTED** [review]
Hello!

I tried to setup a travis test [1] for this but apparently multipathd (I'm not really surprised) doesn't start inside the travis docker container with a minimal config. I would like to test this a bit more (to see how that blockinfile that I also haven't used before) before merging as I haven't touched multipathd in a long time.

[1]: https://travis-ci.org/CSCfi/ansible-role-dell-firmware-upgrade/builds

File: tasks/multipathd.yml (hunk 1 above)
Comment (Klaas-) [reply tasks/multipathd.yml:19]: I have to admit I was not sure what those are needed for; I just used the recommendations from the red hat ticket :)

Comment (Klaas-) [reply tasks/multipathd.yml:19]: My guess is they try to blacklist all devices not needed for the ovirt/rhv node ie all local storage.

Comment (martbhell) [reply tasks/multipathd.yml:19]: That is OK. If it works it works.

**Review by martbhell — COMMENTED** [review]
On my server I don't even have multipathd installed.
Perhaps I can ask you for the details instead?

1. is this multipath bit of the role idempotent? (can we run it twice and it won't change/reload multipathd on the second run)
2. How does the resulting multipath.conf look? Is the multipathd service starting/running without errors?

**Comment by Klaas-** [comment]
I only have multipathd running on the servers that are part of the ovirt/red hat virtualization clusters. So that means they have a multipath.conf and multipathd running preconfigured by that application. I need to add the blacklist parts for dell because dsu does not work with multipathd enabled on the devices it uses during the dsu upgrade process.

Now to your questions:

>     1. is this multipath bit of the role idempotent? (can we run it twice and it won't change/reload multipathd on the second run)

-> yes, it is idempotent. within multipath.conf the block will only be inserted if that block does not exist or was modified outside of ansible; the service will only be reloaded if multipath.conf was changed. One thing I did think about (but have no better solution as of now) is if I could modify an existing blacklist entry directly with something like augeas. But that would mean additional dependencies -- for now I think the solution is allright -- if someone has a pre-existing blacklist block they can PR a better solution or disable mgmt from this role via the switch.

> 
>     2. How does the resulting multipath.conf look? Is the multipathd service starting/running without errors?
-> I can't say that for sure because the default ovirt install is logging a warning during startup; but after adding blacklist there is no additional warnings/errors :) I've used this PR to patch some of my dell servers already so it is working as far as I can tell

**Comment by martbhell** [comment]
Many thanks for the detailed reply and the PR itself :)
Yeah if someone wants different multipath settings using another role sounds reasonable

**Pull request merged**